app/src/main/java/unipd/se18/ocrcamera/ImageProcessing.java

```diff
@@ -0,0 +1,349 @@
+package unipd.se18.ocrcamera;
+
+import android.graphics.Bitmap;
+import android.os.Environment;
+import android.util.Log;
+
+import org.opencv.core.Core;
+import org.opencv.core.Mat;
+import org.opencv.core.MatOfInt4;
+import org.opencv.core.MatOfPoint;
+import org.opencv.core.MatOfPoint2f;
+import org.opencv.core.RotatedRect;
+import org.opencv.core.Size;
+import org.opencv.imgcodecs.Imgcodecs;
+import org.opencv.imgproc.Imgproc;
+import org.opencv.android.Utils;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import static org.opencv.imgproc.Imgproc.INTER_CUBIC;
+import static org.opencv.imgproc.Imgproc.getRectSubPix;
+import static org.opencv.imgproc.Imgproc.getRotationMatrix2D;
+
+
+/**
+ * Class used to analyze the image
+ */
+public class ImageProcessing {
+
+    /*
+        Documentation of the Imgproc class available at:
+        https://docs.opencv.org/java/2.4.2/org/opencv/imgproc/Imgproc.html
+        
+        We referenced a previous instance of the documentation since
+        the newer one is still incomplete
+
+        We are trying to implement all the Imgproc methods,
+        which require a large number of
+        parameters to be called, with a builder statement,
+        to simplify reading and eventually changing them
+        (at the moment there are a little too many "magic numbers")
+     */
+
+    //Tag used to identify the log
+    final String TAG = "openCV";
+
+
```

Comment:
For the actual code style between methods (or in case similar to this) is permitted only one empty line, right?

Reply:
I'll correct it in every method